components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -76,26 +76,17 @@ public Origin getOrigin() {
         return Mono.<Connection>create(sink -> {
             Connection connection = dequeue();
             if (connection != null) {
-                if (borrowedCount.getAndIncrement() < poolSettings.maxConnectionsPerHost()) {
-                    sink.success(connection);
-                } else {
-                    borrowedCount.decrementAndGet();
-                    queueNewConnection(connection);
-                }
+                borrowedCount.incrementAndGet();
+                sink.success(connection);
             } else {
-                if (waitingSubscribers.size() >= poolSettings.maxPendingConnectionsPerHost()) {
+                if (waitingSubscribers.size() < poolSettings.maxPendingConnectionsPerHost()) {
+                    this.waitingSubscribers.add(sink.onDispose(() -> waitingSubscribers.remove(sink)));
```

Comment:
Should we use  doOn operators (eg doFinally() for this one and for the connectionsInEstablishment variable as well) for these  side-effects?

Reply:
Please clarify. Where would we use the `doOn/doFinally` operator?